diff -cr vdr-1.2.1/device.h vdr-1.2.1-dsmcc/device.h
*** vdr-1.2.1/device.h	Sun May 11 09:50:04 2003
--- vdr-1.2.1-dsmcc/device.h	Tue Jun 17 10:05:33 2003
***************
*** 205,212 ****
    cPidHandle pidHandles[MAXPIDHANDLES];
    bool HasPid(int Pid) const;
           ///< Returns true if this device is currently receiving the given PID.
-   bool AddPid(int Pid, ePidType PidType = ptOther);
-          ///< Adds a PID to the set of PIDs this device shall receive.
    void DelPid(int Pid, ePidType PidType = ptOther);
           ///< Deletes a PID from the set of PIDs this device shall receive.
    virtual bool SetPid(cPidHandle *Handle, int Type, bool On);
--- 205,210 ----
***************
*** 217,222 ****
--- 215,223 ----
           ///< Handle->used indicated how many receivers are using this PID.
           ///< Type indicates some special types of PIDs, which the device may
           ///< need to set in a specific way.
+   public:
+   bool AddPid(int Pid, ePidType PidType = ptOther);
+          ///< Adds a PID to the set of PIDs this device shall receive.
  
  // Common Interface facilities:
  
diff -cr vdr-1.2.1/receiver.c vdr-1.2.1-dsmcc/receiver.c
*** vdr-1.2.1/receiver.c	Sun Jul 28 16:14:49 2002
--- vdr-1.2.1-dsmcc/receiver.c	Tue Jun 17 10:06:01 2003
***************
*** 38,43 ****
--- 38,62 ----
    Detach();
  }
  
+ // Hack for dsmcc plugin
+ bool cReceiver::AddPid(int Pid) {
+   int i = 0;
+ 
+   if (device && Pid) {
+      for (; i < MAXRECEIVEPIDS; i++) {
+          if (pids[i] == Pid)
+             return true;
+          if (!pids[i])
+             break;
+      }
+      pids[i] = Pid;
+      device->AddPid(Pid);
+      return true;
+   }
+   return false;
+ }
+ 
+ 
  bool cReceiver::WantsPid(int Pid)
  {
    if (Pid) {
***************
*** 51,56 ****
--- 70,76 ----
    return false;
  }
  
+ 
  void cReceiver::Detach(void)
  {
    if (device)
diff -cr vdr-1.2.1/receiver.h vdr-1.2.1-dsmcc/receiver.h
*** vdr-1.2.1/receiver.h	Sun Jul 28 12:22:01 2002
--- vdr-1.2.1-dsmcc/receiver.h	Tue Jun 17 10:06:45 2003
***************
*** 45,50 ****
--- 45,52 ----
                 // Priority may be any value in the range 0..99. Negative values indicate
                 // that this cReceiver may be detached at any time (without blocking the
                 // cDevice it is attached to).
+   // Add new pid to list. Hack for dsmcc plugin
+   bool AddPid(int Pid);
    virtual ~cReceiver();
    };
